modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/util/MBMessageUtil.java

```diff
@@ -17,19 +17,175 @@
+import com.liferay.portal.kernel.parsers.bbcode.BBCodeTranslatorUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
 import com.liferay.portal.kernel.util.ListUtil;
 import com.liferay.portal.kernel.util.OrderByComparator;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
 import com.liferay.portal.kernel.workflow.WorkflowConstants;
 
 import java.util.Comparator;
 import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang.StringUtils;
 
 /**
  * @author Preston Crary
  */
 public class MBMessageUtil {
 
+	public static String getMessageBody(
+		boolean htmlFormat, MBMessage mbMessage, String quoteMark,
+		ServiceContext serviceContext) {
+
+		if (!htmlFormat) {
+			return _quotedMessage(true, mbMessage.getBody(), quoteMark);
+		}
+
+		if (!mbMessage.isFormatBBCode()) {
+			return mbMessage.getBody();
+		}
+
+		try {
+			String messageBody = BBCodeTranslatorUtil.getHTML(
+				mbMessage.getBody());
+
+			HttpServletRequest httpServletRequest = serviceContext.getRequest();
+
+			if (httpServletRequest == null) {
+				return messageBody;
+			}
+
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)httpServletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			return MBUtil.replaceMessageBodyPaths(themeDisplay, messageBody);
+		}
+		catch (Exception exception) {
+			_log.error(
+				StringBundler.concat(
+					"Unable to parse mbMessage ", mbMessage.getMessageId(),
+					": ", exception.getMessage()));
+		}
+
+		return mbMessage.getBody();
+	}
+
+	public static String getQuoteMark(boolean htmlFormat) {
+		if (htmlFormat) {
+			return StringPool.BLANK;
+		}
+
+		return _QUOTE_MARK + StringPool.SPACE;
+	}
+
+	public static String getRootMessage(
+		boolean htmlFormat, MBMessage mbMessage, String quoteMark,
+		ServiceContext serviceContext) {
+
+		return StringBundler.concat(
+			_getMarkupElement(MarkupElement.START_ROOT, htmlFormat),
+			_getMarkupElement(MarkupElement.START_USER_ROOT, htmlFormat),
+			_getUserName(htmlFormat, mbMessage, quoteMark),
+			_getMarkupElement(MarkupElement.END, htmlFormat),
+			_getMarkupElement(MarkupElement.START_BODY_ROOT, htmlFormat),
+			getMessageBody(htmlFormat, mbMessage, quoteMark, serviceContext),
+			_getMarkupElement(MarkupElement.END, htmlFormat),
+			_getMarkupElement(MarkupElement.END, htmlFormat));
+	}
+
+	public static String getSameLevelMessages(
+		boolean htmlFormat, List<MBMessage> mbMessages, long messageId,
+		String quoteMark, ServiceContext serviceContext) {
+
+		if (ListUtil.isEmpty(mbMessages)) {
+			return StringPool.BLANK;
+		}
+
+		StringBuilder messageSiblings = new StringBuilder(
+			mbMessages.size() + 2);
+
+		messageSiblings.append(
+			_getMarkupElement(MarkupElement.START_SIBLING, htmlFormat));
+
+		for (MBMessage childMessage : mbMessages) {
```

Comment:
Please, make sure the variable names are consistent. If the list is "mbMessages", the variable should be "mbMessage" (no need for the "child" part, as there's no ambiguity here.)